plan/expression_rewriter.go

```diff
@@ -617,7 +648,9 @@ func (er *expressionRewriter) Leave(inNode ast.Node) (retNode ast.Node, ok bool)
 	case *ast.RowExpr:
 		er.rowToScalarFunc(v)
 	case *ast.PatternInExpr:
-		er.inToExpression(v)
+		if v.Sel == nil {
```

Comment:
if v.Sel is not nil, it has been proccessed. I move this check out of the `inToExpression`.